File: app/src/main/java/com/keno/solo/game_logic/boardgame/Boardgame.java
package com.keno.solo.game_logic.boardgame;

import java.util.ArrayList;
import java.util.List;

import com.keno.solo.game_logic.Game;
import com.keno.solo.game_logic.boardgame.BoardPosition;

public abstract class Boardgame<Piece>  extends Game
{
    protected ArrayList<ArrayList<BoardPosition<Piece>>> board;

    public Boardgame
    (
        String name,
        String rating,
        String type,
        String description,
        List<String> genres,
        int price
    )
    {
        super
        (
            name,
            rating,
            type, 
            description,
            genres,
            price
        );
    }

    protected abstract void setBoard();
    public ArrayList<ArrayList<BoardPosition<Piece>>> getBoard()
    {
        return board;
    }
}

File: sources/game_engine/game_logic/boardgame/Boardgame.cpp
#include "game_engine/game_logic/boardgame/Boardgame.hpp"

template <typename BoardPiece>
Boardgame<BoardPiece>::Boardgame
(
    std::string name,
    std::string rating,
    std::string type,
    std::string description,
    std::vector<std::string> genres,
    int price
) : Game
(
    name, 
    rating, 
    type, 
    description, 
    genres, 
    price
)
{
}

template <typename BoardPiece>
std::vector<std::vector<BoardPosition<BoardPiece>>> Boardgame<BoardPiece>::getBoard() const
{
    return board;
}